src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpMethodRepoDAO.scala

```diff
@@ -0,0 +1,32 @@
+package org.broadinstitute.dsde.rawls.dataaccess
+
+import akka.actor.ActorSystem
+import org.broadinstitute.dsde.rawls.model.AgoraEntity
+import org.broadinstitute.dsde.rawls.model.WorkspaceJsonSupport._
+import scala.concurrent.Await
+import scala.concurrent.duration.Duration
+import scala.util.Try
+import spray.client.pipelining._
+import spray.http.HttpCookie
+import spray.http.HttpHeaders.Cookie
+import spray.httpx.SprayJsonSupport._
+
+/**
+ * @author tsharpe
+ */
+class HttpMethodRepoDAO( methodRepoServiceURL: String )( implicit system: ActorSystem ) extends MethodRepoDAO {
+
+  override def getMethodConfig( namespace: String, name: String, version: String, authCookie: HttpCookie ): Option[AgoraEntity] = {
+    val url = s"${methodRepoServiceURL}/configurations/${namespace}/${name}/${version}"
+    import system.dispatcher
+    val pipeline = addHeader(Cookie(authCookie)) ~> sendReceive ~> unmarshal[AgoraEntity]
+    Try(Await.result(pipeline(Get(url)),Duration.Inf)).toOption
```

Comment:
I thought we had decided 3 cases, it is there, it is not there, throw and exception.